pkg/cli/serve/serve.go

```diff
@@ -301,15 +295,22 @@ func (o *serveOptions) Run() error {
 			}
 			r.Use(metrics.PrometheusMiddleware)
 		case config.ClustersService:
-			o.logger.Info("Initialising UpgradePolicy handlers")
-			upgradePolicyHandler := handlers.NewUpgradePoliciesHandler(sdkclient, internalID)
-			// See https://github.com/gorilla/mux#examples
-			r.HandleFunc("/upgrade_policies", upgradePolicyHandler.ServeUpgradePolicyList)
-			r.HandleFunc("/upgrade_policies/{upgrade_policy_id}", upgradePolicyHandler.ServeUpgradePolicyGet)
-			r.HandleFunc("/upgrade_policies/{upgrade_policy_id}/state", upgradePolicyHandler.ServeUpgradePolicyState)
-			o.logger.Info("Initialising Cluster handlers")
-			clusterHandler := handlers.NewClusterHandler(sdkclient, internalID)
-			r.HandleFunc("/", clusterHandler.ServeClusterGet)
+			if !o.fleetMode {
```

Comment:
I'm not 100% about this, but I don't see how this would ever run in fleet mode? It needs the cluster id flag which is only passed for non fleet mode?

Reply:
@typeid we do have plans to eventually use OCM agent as an intermediate proxy in fleet mode, too. See this comment from Ravi: https://redhat-internal.slack.com/archives/C01JJKHC0CF/p1700717837365689?thread_ts=1700714500.668629&cid=C01JJKHC0CF. My preference is to keep it as-is, so that we could use it in the future.

Reply:
Confirmed that the above crashloops OCM-agent with the current main/prod deployment on the rhobs cluster:
```
oc logs ocm-agent-staging-hypershift-6c4557d758-8948q -n openshift-ocm-agent-operator
***** Starting with FIPS crypto enabled *****
time="2024-01-18T06:43:36Z" level=info msg="Starting ocm-agent server"
time="2024-01-18T06:43:36Z" level=info msg="Fleet mode configured" FleetMode=true
time="2024-01-18T06:43:36Z" level=info msg="Start listening on metrics port" Port=8383
time="2024-01-18T06:43:36Z" level=info msg="Connection with OCM initialised successfully in fleet mode"
time="2024-01-18T06:43:37Z" level=error msg="Cluster with external id  not found in OCM database."
time="2024-01-18T06:43:37Z" level=fatal msg="OCM Agent failed to fetch internal cluster ID" error="cluster with external id  not found in OCM database"
```

The reason I had added a check for the fleet-mode there was that it would crash if I didn't: there is no cluster-id flag passed to ocm-agent in fleet mode, so IMO this can't work and something is wrong.

Reply:
Fixed with https://github.com/openshift/ocm-agent/pull/79